compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -804,6 +825,49 @@ object JavaParsers {
       addCompanionObject(statics, cls)
     }
 
+    def recordDecl(start: Offset, mods: Modifiers): List[Tree] =
+      accept(RECORD)
+      val nameOffset = in.offset
+      val name = identForType()
+      val tparams = typeParams()
+      val header = formalParams()
+      val superclass = javaLangRecord() // records always extend java.lang.Record
+      val interfaces = interfacesOpt() // records may implement interfaces
+      val (statics, body) = typeBody(RECORD, name, tparams)
+
+      // We need to generate accessors for every param, if no method with the same name is already defined
+
+      var fieldsByName = header.map(v => (v.name, (v.tpt, v.mods.annotations))).toMap
+
+      for case DefDef(name, paramss, tpt, rhs) <- body
+      if paramss.isEmpty && fieldsByName.contains(name)
+      do
+        fieldsByName -= name
+      end for
+
+      val accessors =
+        (for (name, (tpt, annots)) <- fieldsByName yield
```

Comment:
Hmm `fieldsByName` was constructed using `toMap` which produces a Map whose iteration order isn't guaranteed to be deterministic. Even though we don't emit any bytecode for this, I would prefer if we used a LinkedHashMap to stay deterministic (this might impact error messages for example).

Reply:
I've modified fieldsByName to be a `ListMap`